core/src/main/java/io/questdb/cairo/IntervalPartitionFrameCursorFactory.java

```diff
@@ -49,6 +50,7 @@ public IntervalPartitionFrameCursorFactory(
     ) {
         super(tableToken, metadataVersion, metadata);
         this.timestampIndex = timestampIndex;
+        this.timestampDriver = ColumnType.getTimestampDriver(metadata.getColumnType(timestampIndex));
         this.intervalModel = intervalModel;
```

Comment:
intervalModel and timestampDriver must be aligned at all times. The model should already have the driver. There is no need to keep passing both model and the driver separately. `timestampDriver` must not be a field on the factory and beyond.

Reply:
fixed.